src/Common/FitOnFhir.Common/JwtSecurityTokenHandlerProvider.cs

```diff
@@ -0,0 +1,43 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.IdentityModel.Tokens.Jwt;
+using EnsureThat;
+using Microsoft.IdentityModel.Tokens;
+
+namespace FitOnFhir.Common
+{
+    public class JwtSecurityTokenHandlerProvider : IJwtSecurityTokenHandlerProvider
+    {
+        private JwtSecurityTokenHandler _jwtSecurityTokenHandler;
+
+        /// <inheritdoc/>
+        public void CreateJwtSecurityTokenHandler()
```

Comment:
I think you can just call `_jwtSecurityTokenHandler = new JwtSecurityTokenHandler();` in the constructor of this class, that way you don't need to check if it is null in the other method calls.